datafusion/physical-expr-common/src/physical_expr.rs

```diff
@@ -149,6 +151,11 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug + DynEq + DynHash {
     fn get_properties(&self, _children: &[ExprProperties]) -> Result<ExprProperties> {
         Ok(ExprProperties::new_unknown())
     }
+
+    /// Return the column statistics of this expression given the statistics of the input
```

Comment:
This API seems somewhat overlapping with [`PhysicalExpr::evaluate_bounds`](https://docs.rs/datafusion/latest/datafusion/physical_expr/trait.PhysicalExpr.html#method.evaluate_bounds)

I wonder if there is any way to combine the ideas with the `Statistics` changes @suremarc is looking into in https://github.com/apache/datafusion/pull/13293#issuecomment-2540053084 🤔

Reply:
I wrote up a unified proposal here:
- https://github.com/apache/datafusion/issues/8078#issuecomment-2543895426

Feedback very welcome